### PR TITLE
(PDB-5204) Created script to aggregate Locust results

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

File: ext/bin/locust-load-tests-aggregator [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    usage()
+    exit(0)
+
+new_report_path = sys_args[0]
+sha_timestamp = new_report_path.split('/')[-2:-1]
+aggregated_report_path = '/home/andrei.filipovici/performance_results/locust_aggregated_report.csv'
+
+
+def iterator_to_dict(iterator, key_index, filter = []):
+    # Transforms a iterable object into a dictionary
```
Comment (Filipovici-Andrei) [inline ext/bin/locust-load-tests-aggregator:24]: "an" iterable

Comment (sebastian-miclea) [reply ext/bin/locust-load-tests-aggregator]: Updated. Thanks!

File: ext/bin/locust-load-tests-aggregator [hunk 2]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3.6
+
+import csv, os.path, sys, pry
```
[the remark this answers is not in the document]
Comment (IrimieBogdan) [reply ext/bin/locust-load-tests-aggregator:3]: each import should be on it's own line

**Review by IrimieBogdan — APPROVED** [review]

File: ext/bin/locust-load-tests-aggregator [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        for name, values in self.header.items():
+            data.append([name] + values)
+
+        for name, values in self.body.items():
+            data.append([name] + values)
+
+        for name, values in self.footer.items():
+            data.append([name] + values)
+
+        return data
```
Comment (IrimieBogdan) [inline ext/bin/locust-load-tests-aggregator:47]: Could be refactored
``` python
def to_list(self):
    # Converts header, body and footer to a list of lists representing rows in a csv file
    data = []
    appender(data, self.header)
    appender(data, self.body)
    appender(data, self.footer)

    return data

...

def appender(data, item):
    for name, values in item.items():
        data.append([name] + values)
```

File: ext/bin/locust-load-tests-aggregator [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+
+def usage():
+    usage = """
```
Comment (Filipovici-Andrei) [inline ext/bin/locust-load-tests-aggregator:51]: the usage variable is not needed, you can print the string directly

Comment (sebastian-miclea) [reply ext/bin/locust-load-tests-aggregator]: Updated, thanks!

File: ext/bin/locust-load-tests-aggregator (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebastian-miclea) [reply ext/bin/locust-load-tests-aggregator]: Updated, thanks!

File: ext/bin/locust-load-tests-aggregator [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    def merge(self, csv_report):
+        # Appends a new csv_report object to the current instance,
+        # grouping is done by the key for header, body and footer
+        # csv_report - CsvReport -
+        existing_run_count = 0
+
+        for name, value in self.body.items():
+            existing_run_count = len(self.body[name])
```
[the remark this answers is not in the document]
Comment (sebastian-miclea) [reply ext/bin/locust-load-tests-aggregator]: Yes, I missed that, Thanks!

File: ext/bin/locust-load-tests-aggregator (hunk 3 above)
Comment (sebastian-miclea) [reply ext/bin/locust-load-tests-aggregator]: I've found a simpler way to write it. Have a look.

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**